src/pykeen/triples/triples_factory.py

```diff
@@ -283,9 +284,9 @@ def __init__(
         # Generate relation mapping if necessary
         if relation_to_id is None:
             if self.create_inverse_triples:
-                relation_to_id = create_relation_mapping(
-                    set(self.relation_to_inverse.keys()).union(set(self.relation_to_inverse.values())),
-                )
+                _keys = set(self.relation_to_inverse.keys()) if self.relation_to_inverse else set()
+                _values = set(self.relation_to_inverse.values()) if self.relation_to_inverse else set()
```

Comment:
memory-wise, the new formulation is worse, since the variables `_keys`/`_values`  stay referenced until the method returns.